### ISSUE TITLE
Alias and wait on cy.exec

### ISSUE THREAD

**Issue by nedredmond** [cypress-io/cypress#22924]
### What would you like?

```
cy.exec('my command that takes forever to run').as('backgroundTask'); // the script begins execution runs in the background

// do things that do not require task to be complete
...

// now I need this background task to have finished to proceed
cy.wait('@backgroundTask').its('stdout').then(console.log);
cy.fixture('generatedByScript').its('data.i.need').then(console.log);

### Why is this needed?

Some setup scripts take a long time to run. Long test runtimes slow down CI pipelines. It would be helpful to fire off a script and allow the test to continue running until I needed the result of it. Or, if I don't need its result, just allow it to run in the background without interrupting the flow of the test.

### Other

_No response_

**Comment by jennifer-shehane** [comment]
The underlying need for this is a good one: kick off slow setup work, keep the test moving, and only block once you actually need the result.

cy.exec() was deprecated in [Cypress 15.21.0](https://docs.cypress.io/app/references/changelog#15-21-0) and will be removed in a future major release, so we won't be addressing this directly. We suggest moving over to [cy.task()](https://docs.cypress.io/api/commands/task), which runs in the Node process that already loads your Cypress config instead of going through the shell on the machine running Cypress.

Aliasing with .as() and blocking with cy.wait('@alias') is not something we plan to add to cy.task() either. cy.wait() waits on routes and on time, and adding a third meaning would make it harder to reason about, not easier. But the pattern you want is a few lines in Node, and you get to control exactly what "start" and "await" mean:

```js
// cypress.config.js
const { defineConfig } = require('cypress')
const { execFile } = require('child_process')
const { promisify } = require('util')

const execFileAsync = promisify(execFile)
const pending = new Map()

module.exports = defineConfig({
  e2e: {
    setupNodeEvents (on, config) {
      on('task', {
        startBackgroundTask (name) {
          // Kick off the work and return immediately so the test keeps running
          pending.set(name, execFileAsync('scripts/slow-setup.sh', [name], {
            cwd: config.projectRoot,
          }))

          return null
        },

        async awaitBackgroundTask (name) {
          const promise = pending.get(name)

          if (!promise) {
            throw new Error(`No background task started for "${name}"`)
          }

          pending.delete(name)

          const { stdout, stderr } = await promise

          return { stdout, stderr }
        },
      })
    },
  },
})
```

```js
// in your spec
cy.task('startBackgroundTask', 'seedData')

// do things that do not need the task to be finished
cy.visit('/')
cy.get('[data-cy="login"]').click()

// now block until it is done
cy.task('awaitBackgroundTask', 'seedData').its('stdout').then(cy.log)
cy.fixture('generatedByScript').its('data.i.need').then(cy.log)
```

We're going to close this issue, since the command it asks for is on its way out and the behavior is available today through cy.task(). If this pattern does not cover your case, open a new issue.
